lib/heroku-deflater/cache_control_manager.rb

```diff
@@ -0,0 +1,37 @@
+module HerokuDeflater
+  class CacheControlManager
+    attr_reader :app,:max_age
+
+    def initialize(app)
+      @app = app
+    end
+
+    def setup_max_age(max_age)
+      @max_age = max_age
+      if rails_version_5?
+        app.config.public_file_server.headers ||= {}
+        app.config.public_file_server[:headers] = {'Cache-Control' => cache_control}
```

Comment:
You have a conditional assignment a line before, and here you replace it with a new content. Leftover?